usecases/scheduled_execution/run_scheduled_execution.go

```diff
@@ -261,231 +231,239 @@ func (usecase *RunScheduledExecution) executeScheduledScenario(ctx context.Conte
-		executionScenario := func(ctx context.Context, object models.ClientObject, i int) error {
-			ctx, span := tracer.Start(
-				ctx,
-				"Batch RunScheduledExecution.executeScheduledScenario",
-				trace.WithAttributes(
-					attribute.String("scenario_id", scenario.Id),
-					attribute.Int64("object_index", int64(i)),
-					attribute.String("object_type", scenario.TriggerObjectType),
-				))
-			defer span.End()
-			scenarioExecution, err := evaluate_scenario.EvalScenario(
+	decisionsToCreate, err := usecase.repository.StoreDecisionsToCreate(ctx, exec, models.DecisionToCreateBatchCreateInput{
+		ScheduledExecutionId: scheduledExecutionId,
+		ObjectId:             objectIds,
+	})
+	if err != nil {
+		return false, numbersOfDecisions{}, err
+	}
+
+	var nbEvaluatedDec, nbCreatedDec int
+	for i, decisionToCreate := range decisionsToCreate {
+		created, err := usecase.createSingleDecisionForObjectId(
+			ctx,
+			db,
+			decisionToCreate,
+			scenario,
+			dataModel,
+			scheduledExecutionId,
+			pivot,
+			i,
+		)
+		nbEvaluatedDec += 1
+		if err != nil {
+			updateErr := usecase.repository.UpdateDecisionToCreateStatus(
 				ctx,
-				evaluate_scenario.ScenarioEvaluationParameters{
-					Scenario:     scenario,
-					ClientObject: object,
-					DataModel:    dataModel,
-					Pivot:        pivot,
-				},
-				evaluate_scenario.ScenarioEvaluationRepositories{
-					EvalScenarioRepository:     usecase.repository,
-					ExecutorFactory:            usecase.executorFactory,
-					IngestedDataReadRepository: usecase.ingestedDataReadRepository,
-					EvaluateAstExpression:      usecase.evaluateAstExpression,
-					SnoozeReader:               usecase.snoozesReader,
-				},
+				exec,
+				decisionToCreate.Id,
+				models.DecisionToCreateStatusFailed,
 			)
+			utils.LogAndReportSentryError(ctx, errors.Join(err, updateErr))
+			// Stop at the first encountered error, store the number of created decisions on the scheduled execution
+			return false, numbersOfDecisions{
+				evaluated: nbEvaluatedDec,
+				created:   nbCreatedDec,
+			}, nil
+		}
+		if created {
+			nbCreatedDec += 1
+		}
 
-			if errors.Is(err, models.ErrScenarioTriggerConditionAndTriggerObjectMismatch) {
-				logger := utils.LoggerFromContext(ctx)
-				logger.InfoContext(ctx,
-					fmt.Sprintf("Trigger condition and trigger object mismatch: %s", err.Error()),
-					"scenarioId", scenario.Id,
-					"object_index", i,
-					"triggerObjectType", scenario.TriggerObjectType,
-					"object", object)
-				return nil
-			} else if err != nil {
-				return errors.Wrapf(err, "error evaluating scenario in executeScheduledScenario %s", scenario.Id)
-			}
-
-			decision := models.AdaptScenarExecToDecision(scenarioExecution, object, &scheduledExecutionId)
-			err = usecase.decisionRepository.StoreDecision(
+		// Update the number of created decisions on the scheduled execution, but not at every decision
+		if i%100 == 0 {
```

Comment:
This means we'll have an intermediate estimate of the number of decisions evaluated/created during the batch execution. The final number will be written at the end of the job.
Problem: this logic is not directly reusable once I move to an async worker implementation of batch decision creation (I can then no longer rely on the sequence of decisions being created).
I'll think of an evolution, worst case I'll drop this "feature"